### PR TITLE
Fix debuglogs.org integration

### PR DESCRIPTION
**Pull request by michaelkirk**
content-type is now not only available, but required.

### CODE REVIEW

File: Scripts/debug_log_upload.py [hunk 1]
```diff
@@ -41,24 +44,30 @@ def execute_command(command):
...
     for field_name in upload_fields:
-        field_value = upload_fields[field_name]
-        curl_command.append('-F')
-        curl_command.append("'%s=%s'" % (field_name, field_value, ))
+        add_field(curl_command, field_name, upload_fields[field_name])
+
+    add_field(curl_command, "content-type", "application/octet-stream")
+ 
     curl_command.append('-F')
-    curl_command.append("'file=@%s'" % (args.file,))
```
Comment (michaelkirk) [inline Scripts/debug_log_upload.py]: This actually didn't work with my version of curl.

File: Scripts/debug_log_upload.py [hunk 2]
```diff
@@ -41,24 +44,30 @@ def execute_command(command):
...
     upload_key = upload_key + os.path.splitext(args.file)[1]
-    upload_fields['key'] = upload_key
     
     download_url = 'https://debuglogs.org/' + upload_key
     print 'download_url:', download_url
     
     curl_command = ['curl', '-v', '-i', '-X', 'POST']
+
+    # key must appear before other fields
+    add_field(curl_command, 'key', upload_key)
```
Comment (michaelkirk) [inline Scripts/debug_log_upload.py]: aws requirement, else you get some 4xx response.

File: SignalMessaging/utils/OWSPreferences.m [hunk 3]
```diff
@@ -255,6 +255,7 @@ - (void)applyCallLoggingSettingsForLegacyUsersWithTransaction:(YapDatabaseReadWr
         }
     }();
 
+    DDLogInfo(@"%@ Migrating setting - System Call Enabled: %d", self.logTag, shouldLogCallsInRecents);
```
Comment (michaelkirk) [inline SignalMessaging/utils/OWSPreferences.m]: unrelated, but wanted to get this in.

File: Signal/src/util/Pastelog.m [hunk 4]
```diff
@@ -150,6 +151,8 @@ - (void)uploadFileWithUploadUrl:(NSString *)uploadUrl fields:(NSDictionary *)fie
                 NSString *fieldValue = fields[fieldName];
                 [formData appendPartWithFormData:[fieldValue dataUsingEncoding:NSUTF8StringEncoding] name:fieldName];
             }
+            [formData appendPartWithFormData:[OWSMimeTypeApplicationZip dataUsingEncoding:NSUTF8StringEncoding] name:@"content-type"];
```
Comment (charlesmchen) [inline Signal/src/util/Pastelog.m]: This should be `self.mimeType`, not `OWSMimeTypeApplicationZip`.

Comment (charlesmchen) [reply Signal/src/util/Pastelog.m]: or probably `weakSelf.mimeType`.

Comment (michaelkirk) [reply Signal/src/util/Pastelog.m]: Done, thanks!

**Pull request merged**